internals/daemon/api_notices.go

```diff
@@ -92,12 +128,50 @@ func v1GetNotices(c *Command, r *http.Request, _ *UserState) Response {
 		notices = st.Notices(filter)
 	}
 
-	if notices == nil {
-		notices = []*state.Notice{} // avoid null result
-	}
 	return SyncResponse(notices)
 }
 
+// Get the UID of the request. If the UID is not known, return an error.
+func uidFromRequest(r *http.Request) (uint32, error) {
+	_, uid, _, err := ucrednetGet(r.RemoteAddr)
+	if err != nil {
+		return 0, fmt.Errorf("could not parse request UID")
+	}
+	return uid, nil
+}
+
+// Construct the user IDs filter which will be passed to state.Notices.
+func sanitizeUserIDFilter(queryUserID []string) (*uint32, error) {
+	userIDStrs := strutil.MultiCommaSeparatedList(queryUserID)
+	if len(userIDStrs) > 1 {
```

Comment:
I *think* if you say `user-id= ` (only spaces) you could get `len(userIDStrs)==0` here. So to avoid a panic in that case we should say `if len(userIDStrs) != 1 { ... }` here and say something like `must only include one "user-id"`.

Reply:
Good point. We need to make sure we only call `sanitizeUserIDFilter` if `len(query["user-id"]) > 0` then (which we currently do, but the other `sanitize*` functions don't assume this).

I'm still not sure how `MultiCommaSeparatedList` could return a length 0 list of strings though, in this case... if given `"user-id"=" "`, wouldn't it end up with `[]string{" "}`?

If the query argument is a zero-length string, then would `MultiCommaSeparatedList` turn this into a zero-length slice of strings, or a one-length slice `[]string{""}`? I think that's what it comes down to. It shouldn't treat `" "` as any different from any other one-length string, I would think.

Reply:
I checked, you're right, `CommaSeparatedList` throws away empty strings. Since it returns a value which is then overrides the `userID` previously set, I'd rather not throw an error if `queryUserID` has length 0 and then need to do extra error handling on the caller side. I'd rather just add the comment saying not to call `sanitizeUserIDFilter` if `queryUserID` has length 0.